sha512/src/main/kotlin/org/komputing/khash/sha512/Sha512.kt

```diff
@@ -199,65 +129,45 @@ object Sha512 {
     private fun padMessage(input: ByteArray): ByteArray {
         // Need to append at least 17 bytes (16 for length of the message, and 1 for the '1' bit)
         // then fill with 0s until multiple of 128 bytes
-        var size = input.size + 17
-
-        val remainder = size % 128
-        if (remainder != 0) {
-            size += 128 - remainder
+        val size = (input.size + 17).let {
+            if (it % 128 == 0) it else it + 128 - it % 128
```

Comment:
I thought I could save one division here.

Reply:
Yea  I strongly wanted the val here - might have wanted it too much